editing_and_imputation/editing/thousand_pounds_correction/technical_specification.md

```diff
@@ -0,0 +1,206 @@
+by the method
+* Thresholds set are a good indication of whether a value should be corrected
+* Although predictive and auxiliary are both optional, at least one has to be
+provided for the calculation, else a method error is produced and the TPC
+marker "S" is returned in the output.
+* The unique_identifier is unused directly by the method and is passed-through
+as-is into the output dataset. This attribute is provided to allow a user
+context to be provided as required.
+* If no target_variables are provided only the principal_variable may
+be adjusted
+
+## 4.0 Data records
+
+The following field types will be present in the input and
+output records, for more details see the methodology specification.
+
+### 4.1 Input records
+
+* Unique Identifier – String
+* principal Variable – Single variable, numeric
+* Target Variable(s) – List of monetary variables that may be automatically corrected
+* Predictive Variable – Single variable, numeric – nulls allowed
+* Auxiliary Variable – Optional, numeric – nulls allowed
+* Upper Limit – Single variable, numeric, must be greater than Lower Limit
+* Lower Limit – Single variable, numeric, must be less than Upper Limit
+
+### 4.2 Output records
+
+* Unique Identifier – Business Reporting Unit
+* TPC Ratio – String
+* Final principal Variable – String
+* Final Target Variables – Can be a list, string – nulls allowed
+* TPC Marker – To indicate the result of the Thousand Pounds Correction
+method, string
+
+## 5.0 Process Overview
+
+The following is a key of useful formula definitions/assumptions
+
+* unique_identifier: (String) - Unique identifier
+* principal_variable: (Float) - Numeric value that the method is working on
+* predictive: (Float) - Optional - Numeric value used for comparison.
+A previous 'valid' value (i.e. Returned/Imputed/Constructed)
+* auxiliary: (Float) - Optional - Alternative numeric` value used when
+a predictive value is not available and required by the user
+* upper_limit: (Float) - Upper bound of 'error value' threshold
+* lower_limit: (Float) - Lower bound of 'error value' threshold
+* target_variables: (List of Variables) - Optional - List of linked
+question and values to potentially be adjusted
+
+### 5.1 Validate Data Input (Stage 1)
+
+The methodology specification says that the method checks if
+both predictive and auxiliary are absent. If this is the case validation
+will raise an thousand pounds exception
+and method processing will stop. This is seen by the formulas below
+
+```bash
+    x_{predictive} = None
+```
+
+and
+
+```bash
+    x_{auxiliary} = None
+```
+
+we can alternatively display this as below
+
+```bash
+    predictive = None
+```
+
+and
+
+```bash
+    auxiliary = None
+```
+
+If the predictive and auxiliary are None then we return a tailored
+error message. However, if both are 0 or predictive is 0 and
+auxiliary is none then and stop the method with an "S" TPC marker.
```

Comment:
also an S is thrown if predictive is none and aux is 0

Reply:
Done